### PR TITLE
Fix in-place input normalization when using `sentence`/`speaker` norm

### LINKED ISSUE #2390: spkrec-ecapa-voxceleb-mel-spec model modifies mel spectrum in place when used with CPU
### Describe the bug

Thanks for your wonderful project. I would love to make my first speechbrain contribution and help fix this issue :). when I encode a mel spectrogram using a CPU using MelSpectrogramEncoder, the underlying mel spectrogram data is modified

### Expected behaviour

Calculating the embedding should not modify the original data

### To Reproduce

test_in_place.py
```python
from speechbrain.pretrained import MelSpectrogramEncoder
import torch
from torch.nn.functional import mse_loss
import torchaudio

# On CPU
waveform, sr = torchaudio.load("tests/samples/TTS/LJ050-0131.wav")
waveform = torchaudio.functional.resample(waveform, sr, 16000)
spk_emb_encoder = MelSpectrogramEncoder.from_hparams(source="speechbrain/spkrec-ecapa-voxceleb-mel-spec", savedir="spk_emb_encoder_checkpoints", run_opts={"device": "cpu"})
mel = spk_emb_encoder.mel_spectogram(waveform)
mel_copy = torch.clone(mel)
spkr_emb = spk_emb_encoder.encode_mel_spectrogram(mel_copy)
print(mse_loss(mel, mel_copy))

# On CUDA device
waveform, sr = torchaudio.load("tests/samples/TTS/LJ050-0131.wav")
waveform = torchaudio.functional.resample(waveform, sr, 16000)
spk_emb_encoder = MelSpectrogramEncoder.from_hparams(source="speechbrain/spkrec-ecapa-voxceleb-mel-spec", savedir="spk_emb_encoder_checkpoints", run_opts={"device": "cuda"})
mel = spk_emb_encoder.mel_spectogram(waveform)
mel_copy = torch.clone(mel)
spkr_emb = spk_emb_encoder.encode_mel_spectrogram(mel_copy)
print(mse_loss(mel, mel_copy))
```
python tests/test_in_place.py 
tensor(33.1424)
tensor(0.)

### Environment Details

speechbrain '0.5.16'

### Relevant Log Output

_No response_

### Additional Context

It only happens with CPU

### PR DESCRIPTION
**Pull request by asumagic**
## What does this PR do?

Fixes #2390

`InputNormalization`, when using the `"sentence"`/`"speaker"` normalization, would do sliced assignments over the input `x` tensors, thus potentially mutating data the user has passed. This results in bugs like #2390.

Remaining questions before merging:

- Q: Do we actually ever backpropagate through InputNormalization?
  - A: Not often, apparently: `requires_grad` must be `True` but it defaults to `False`.
- Q: If so, why didn't it ever break due to the slice assignment?
  - A: Need to check, but it probably does. 
- Q: If so, we should check if this solves it (but from my toying around the backpropagation in PyTorch I think it should).
  - A: Should be OK, but didn't test it in training.

<details>
  <summary><b>Before submitting</b></summary>

- [x] Did you read the [contributor guideline](https://speechbrain.readthedocs.io/en/latest/contributing.html)?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing [tests](https://github.com/speechbrain/speechbrain/tree/develop/tests) pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Does your code adhere to project-specific code style and conventions?

</details>

## PR review

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified
- [ ] Confirm that the changes adhere to compatibility requirements (e.g., Python version, platform)
- [ ] Review the self-review checklist to ensure the code is ready for review

</details>

### CODE REVIEW

**Review by Adel-Moumen — APPROVED** [review]
LGTM. Thanks @asumagic :)
